lib/solarwinds_apm/support/otlp_endpoint.rb

```diff
@@ -0,0 +1,196 @@
+        true
+      end
+    end
+
+    # token and endpoint also need to be considered for get settings
+    # endpoint is for get settings, populate to OTEL_EXPORTER_OTLP_METRICS_ENDPOINT at the end for otlp related exporter
+    # three sources: otlp env variable, sw env variable, sw config file
+
+    # sampler_config = {
+    #         collector: "https://#{ENV.fetch('SW_APM_COLLECTOR', 'apm.collector.na-01.cloud.solarwinds.com')}:443",
+    #         service: service_key_name[1],
+    #         headers: "Bearer #{service_key_name[0]}",
+    #         tracing_mode: SolarWindsAPM::Config[:tracing_mode],
+    #         trigger_trace_enabled: SolarWindsAPM::Config[:trigger_tracing_mode],
+    #         transaction_settings: SolarWindsAPM::Config[:transaction_settings]
+    #       }
+    def configure_otlp_endpoint(data_type)
+      # for staging, our purpose, just use OTEL_EXPORTER_OTLP_METRICS_ENDPOINT directly
+      # https://otel.collector.cloud.solarwinds.com:443/v1/traces
+      # SW_ENDPOINT_REGEX = /^apm\.collector(?:\.[a-z0-9-]+)?\.cloud\.solarwinds\.com$/
+
+      return unless %w[TRACES METRICS LOGS].include?(data_type)
+
+      data_type_upper = data_type.upcase
+      data_type = data_type.downcase
+      endpoint_type = if ENV["OTEL_EXPORTER_OTLP_#{data_type_upper}_ENDPOINT"]
+                        "#{data_type}_endpoint"
+                      elsif ENV['OTEL_EXPORTER_OTLP_ENDPOINT']
+                        'general_endpoint'
+                      elsif ENV['SW_APM_COLLECTOR'].nil? && !@lambda_env
+                        'default_nil'
+                      elsif ENV['SW_APM_COLLECTOR'].to_s.match?(SW_ENDPOINT_REGEX)
+                        'apm_proto'
+                      else
+                        'invalid'
+                      end
+
+      SolarWindsAPM.logger.debug { "[#{self.class}/#{__method__}] endpoint_type: #{endpoint_type}" }
+
+      # endpoint = nil
+      case endpoint_type
+      when "#{data_type}_endpoint" || 'general_endpoint'
+        # no need to worry about metrics endpoint, just need to make sure the collector endpoint is set for getsetting
+        endpoint = endpoint_type == 'general_endpoint' ? ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil) : ENV.fetch("OTEL_EXPORTER_OTLP_#{data_type_upper}_ENDPOINT", nil)
+
+        if endpoint.to_s.match?(OTEL_ENDPOINT_REGEX)
+          matches = endpoint.to_s.match(OTEL_ENDPOINT_REGEX)
+          region = matches[1]
+          sampler_collector_endpoint = DEFAULT_APMPROTO_ENDPOINT.gsub('na-01', region)
+          ENV['SW_APM_COLLECTOR'] = sampler_collector_endpoint
+          # if all special OTEL_ENDPOINT, then SW_APM_COLLECTOR will fall back to default DEFAULT_APMPROTO_ENDPOINT
+        end
+
+      when 'default_nil'
+        # default_nil => no otlp endpoint or no SW_APM_COLLECTOR, use the default apm proto endpoint
```

Comment:
```suggestion
        # default_nil => no OTLP endpoint and no SW_APM_COLLECTOR, use the fallback values
```

Reply:
Given that we only end up here if both OTLP endpoint and SW APM collector config are not set, there needs to be logic somewhere that handles the case:

* OTLP exporter endpoint explicitly set
* SW APM collector config not set

expected behaviour is OTLP exporter endpoint resolved to whatever is explicitly set, while SW APM collector config defaults to the fallback value `apm.collector.na-01.cloud.solarwinds.com`. I think this `OTLPEndPoint` class in general would be clearer as `ResolveEndpoint` class that resolved the "settings" vs "exporter" endpoints separately.